apps/wallet-mobile/src/features/DappExplorer/useCases/WebBrowserScreen.tsx

```diff
@@ -0,0 +1,157 @@
+import {useFocusEffect} from '@react-navigation/native'
+import React, {useState} from 'react'
+import {Platform, Share, StyleSheet, TouchableOpacity, View} from 'react-native'
+import {WebView} from 'react-native-webview'
+
+import {TextInput} from '../../../components'
+import {useSelectedWallet} from '../../../SelectedWallet'
+import {useConnectWalletToWebView} from '../common/hooks'
+import {
+  NavigationArrowLeftIcon,
+  NavigationArrowRightIcon,
+  NavigationRefreshIcon,
+  NavigationShareIcon,
+} from '../common/icons'
+
+const DAPP_URL = 'https://app.dexhunter.io/'
+
+const ENABLED_BUTTON_COLOR = '#383E54'
+const DISABLED_BUTTON_COLOR = '#8A92A3'
+
+export const WebBrowserScreen = () => {
+  const [sessionId] = useState(() => Math.random().toString(36).substring(7))
+
+  return (
+    <View style={styles.root}>
+      <TabScreen sessionId={sessionId} />
+    </View>
+  )
+}
+
+const styles = StyleSheet.create({
+  root: {
+    flex: 1,
+  },
+  navigationContainer: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    height: 32,
+    paddingVertical: 16,
+    paddingHorizontal: 7,
+  },
+  navigationButton: {
+    flex: 1,
+    width: 24,
+    height: 24,
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+})
+
+function TabScreen({sessionId}: {sessionId: string}) {
+  const ref = React.useRef<WebView | null>(null)
+  const wallet = useSelectedWallet()
+  const {handleEvent, initScript} = useConnectWalletToWebView(wallet, ref, sessionId)
+  const [currentUrl, setCurrentUrl] = useState(DAPP_URL)
+  const [inputUrl, setInputUrl] = useState('')
+
+  const [canGoBack, setCanGoBack] = React.useState(false)
+  const [canGoForward, setCanGoForward] = React.useState(false)
+
+  const [isFocused, setIsFocused] = useState(false)
+
+  useFocusEffect(() => {
+    ref.current?.injectJavaScript(initScript)
+    setIsFocused(true)
+    return () => setIsFocused(false)
+  })
+
+  const handleReloadPress = () => {
+    ref.current?.reload()
+  }
+
+  const handleBackPress = () => {
+    ref.current?.goBack()
+  }
+
+  const handleForwardPress = () => {
+    ref.current?.goForward()
+  }
+
+  const handleSharePress = () => {
+    if (Platform.OS === 'android') {
+      Share.share({message: currentUrl})
+    } else {
+      Share.share({url: currentUrl})
+    }
+  }
+
+  const handleUrlSubmit = (url: string) => {
+    const normalizedUrl = url.startsWith('http') ? url : `https://${url}`
+    setCurrentUrl(normalizedUrl)
+  }
+
+  const handleLoadingEnd = (url: string) => {
+    setInputUrl(url)
+    setCurrentUrl(url)
+    ref.current?.injectJavaScript(initScript)
+  }
+
+  return (
+    <View style={{flex: 1}}>
```

Comment:
if it's a screen it must be wrapped within the SafeAreaView, here, or at the Navigator.

Reply:
na, if it is on nav better, we just can't navigate the user off, otherwise it will wrap twice the edges.